tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,553 @@
+}
+
+void aics_desc(const char *description, uint8_t len)
+{
+	int err;
+
+	LOG_DBG("AICS description");
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		err = bt_aics_description_set(included.aics[i], description);
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_AICS, AICS_DESCRIPTION,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+			return;
+		}
+	}
+	tester_rsp(BTP_SERVICE_ID_AICS, AICS_DESCRIPTION, CONTROLLER_INDEX,
+		   BTP_STATUS_FAILED);
+}
+
+void tester_handle_aics(uint8_t opcode, uint8_t index, uint8_t *data, uint16_t len)
+{
+	switch (opcode) {
+	case AICS_READ_SUPPORTED_COMMANDS:
+		aics_supported_commands(data, len);
+		break;
+	case AICS_SET_GAIN:
+		aics_set_gain(data, len);
+		break;
+	case AICS_MUTE:
+		aics_mute();
+		break;
+	case AICS_UNMUTE:
+		aics_unmute();
+		break;
+	case AICS_MAN_GAIN:
+		aics_man_gain();
+		break;
+	case AICS_AUTO_GAIN:
+		aics_auto_gain();
+		break;
+	case AICS_MAN_GAIN_ONLY:
+		aics_auto_gain_only();
+		break;
+	case AICS_AUTO_GAIN_ONLY:
+		aics_auto_gain_only();
+		break;
+	case AICS_DESCRIPTION:
+		aics_desc(data, len);
+		break;
+	case AICS_MUTE_DISABLE:
+		aics_mute_disable();
+		break;
+	default:
+		tester_rsp(BTP_SERVICE_ID_AICS, opcode, index,
+			   BTP_STATUS_UNKNOWN_CMD);
+		break;
+	}
+}
+
+/* Volume Offset Control Service */
+static void vocs_supported_commands(uint8_t *data, uint16_t len)
+{
+	struct net_buf_simple *buf = NET_BUF_SIMPLE(BTP_DATA_MAX_SIZE);
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_u8(buf, VOCS_READ_SUPPORTED_COMMANDS);
+	net_buf_simple_add_u8(buf, VOCS_UPDATE_LOC);
+	net_buf_simple_add_u8(buf, VOCS_UPDATE_DESC);
+
+	tester_send(BTP_SERVICE_ID_VOCS, VOCS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, buf->data, buf->len);
+}
+
+static void vocs_state_cb(struct bt_vocs *inst, int err, int16_t offset)
+{
+	LOG_DBG("VOCS state callback");
+}
+
+static void vocs_location_cb(struct bt_vocs *inst, int err, uint32_t location)
+{
+	LOG_DBG("VOCS location callback");
+}
+
+static void vocs_description_cb(struct bt_vocs *inst, int err,
+				char *description)
+{
+	LOG_DBG("VOCS desctripion callback");
+}
+
+static struct bt_vocs_cb vocs_cb = {
+	.state = vocs_state_cb,
+	.location = vocs_location_cb,
+	.description = vocs_description_cb
+};
+
+void vocs_audio_desc(const char *description, uint8_t len)
+{
+	int err;
+
+	LOG_DBG("VOCS description");
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		err = bt_vocs_description_set(included.vocs[i], description);
+		if (err) {
+			tester_rsp(BTP_SERVICE_ID_VOCS, VOCS_UPDATE_DESC,
+				   CONTROLLER_INDEX, BTP_STATUS_FAILED);
+			return;
+		}
+	}
+	tester_rsp(BTP_SERVICE_ID_VOCS, VOCS_UPDATE_DESC, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+void vocs_audio_loc(uint8_t *data, uint8_t len)
+{
+	int err;
+
+	LOG_DBG("VOCS location");
+	for (int i = 0; i < CONFIG_BT_VCP_VOL_REND_AICS_INSTANCE_COUNT; i++) {
+		err = bt_vocs_location_set(included.vocs[i], (int)data);
```

Comment:
```suggestion
		err = bt_vocs_location_set(included.vocs[i], (uint32_t)data);
```

Perhaps you should implement this as a `cmd` as well, as you've done for the others?